### PR TITLE
pytest: Allow default test file naming patterns inside tests folders

### PR DESCRIPTION
**Pull request by echoix**
To help out https://github.com/OSGeo/grass/pull/4755 and also myself, that I worked-around in https://github.com/OSGeo/grass/pull/4700 by renaming the file (p.s.: still waiting for feedback on the expected behaviour there)

This PR allows to use file names like shown in pytest docs here 
https://docs.pytest.org/en/latest/reference/reference.html#confval-python_files, as long as it is inside a tests folder instead of anywhere. The only expected file name pattern currently is not as natural to me, requiring to end with `_test.py`.

### CODE REVIEW

**Comment by echoix** [comment]
Oh, there are two tests that fail, that weren't collected before due to the expected behaviour not being the one implemented.

**Comment by echoix** [comment]
I found the reason why the test added in https://github.com/OSGeo/grass/pull/3582, doesn't work:

https://github.com/OSGeo/grass/blob/856a5141ea392b620e23dbe486e595564e2fb629/python/grass/script/tests/test_script_task.py#L1-L11

It works when pytest is run inside a grass shell, where GISBASE, GISRC are set, and the modules, like "r.mapcalc.simple" are in path (the bin and scripts subfolders of the install location are in path).


So, I used a fixture that creates a temporary session like used in other pytest tests, but it wasn't enough, as grassTask doesn't take the "env" argument. For this, I used monkeypatch to set the env vars and undo once done: https://docs.pytest.org/en/latest/how-to/monkeypatch.html#monkeypatching-environment-variables

That makes it working again (for the first time in CI), as it wasn't run.

To ensure that modules in both the script and bin folders are usable, I added two other tests. The d.vect one is taken from one of the few usages of grassTask in the source code, in the gui.

**Comment by echoix** [comment]
Only the `GISBASE`, `GISRC` and `PATH` were needed to monkeypatch to get these tests working. But, to keep it general and easier to use elsewhere, I kept the full loop, even if almost nothing was different before and inside the session (except `PATH`, obviously, but also `LD_LIBRARY_PATH`).

**Comment by echoix** [comment]
@Shreshth-Malik Take a look at the workaround I used here. Maybe it could be useful for you too if you got functions that don't work with the env from a `SessionHandle`.

**Comment by Shreshth-Malik** [comment]
@echoix Thank you for this. I will definitely take a look and try to put it in my tests

**Comment by echoix** [comment]
Can someone take a quick look at this? The original intent of the PR was to have pytest run on tests for (new) users following the pytest docs. I have asked someone else to do the change, that I would have approved immediately, but there were some new tests failures that needed to be solved. So I did them myself, so I need to be approved by someone else. I would've approved this PR if it was somebody else that did it. To help out keep the trace of what was done, I document above in the comment the thoughts and the solution. That's what is important. There is one person that is more knowledgeable on this subject precise subject, but we cannot rely on only one person to be always available, effectively having a bus factor of 1 on reviewing testing infrastructure.

With the documentation above, I still recommend merging this even if not scrutated as much, so PRs of other new users can go through, and let them work on new tests. It fixes a real overlook of previous PRs that were merged without having their tests actually run, thus not failing.

File: python/grass/script/tests/test_script_task.py
Comment (neteler) [inline python/grass/script/tests/test_script_task.py:13]:
```suggestion
    This allows functions not accepting an env dictionary argument to work in tests"""
```

File: python/grass/script/tests/test_script_task.py
Comment (neteler) [inline python/grass/script/tests/test_script_task.py:34]: I am wondering if using a `d.*` command in this setting is a good idea - will it not lead to

`ERROR: Neither MONITOR (managed by d.mon command) nor GRASS_RENDER_IMMEDIATE (used for direct rendering) defined`
?

Comment (echoix) [reply python/grass/script/tests/test_script_task.py:34]: If I understand that part correctly, on what grassTask and that code snippet does, is that it plays more with the parser and what options are possible, rather than actually running it.

If it had an error, I think we would've seen it in there

File: python/grass/script/tests/test_script_task.py
Comment (echoix) [reply python/grass/script/tests/test_script_task.py:13]: Oops, totally right. I didn't realize it, as in French, "dictionnaire" has two Ns, so it didn't look wrong when reading it.

**Comment by echoix** [comment]
I'm still hoping for anyone to approve this PR to unblock others

File: python/grass/script/tests/test_script_task.py
Comment (pesekon2) [inline python/grass/script/tests/test_script_task.py:55]: Wouldn't this look better if we put the sides (or at least the right side) of the equation into variables? This double-lined equation where the second line starts with `==` looks absolutely terrible in my opinion and using two (or one) variables would make it more readable. 

However, as it might be just my opinion and the black linter likes it, I leave it up to your consideration and I could live with this being in the code. Looks fine otherwise, approving.

**Review by pesekon2 — APPROVED** [review]

File: python/grass/script/tests/test_script_task.py
Comment (pesekon2) [reply python/grass/script/tests/test_script_task.py]: Resolved in [ab92ade](https://github.com/OSGeo/grass/pull/4784/commits/ab92ade4e81b9d37881dddf6be916e82a8752b6e)

**Comment by pesekon2** [comment]
Thanks @echoix. Approving.

**Review by pesekon2 — APPROVED** [review]

**Pull request merged**
